src/components/schemas/dynamic-list.ts

```diff
@@ -32,3 +29,19 @@ export interface BeagleListViewInterface {
   key?: string,
   __suffix__?: string,
 }
+
+export interface BeagleGridViewInterface extends DynamicListBaseInterface {  
+  numColumns?: number,
+}
+
+export type Direction = 'VERTICAL' | 'HORIZONTAL'
+
+export interface BeagleListViewInterface extends DynamicListBaseInterface{
+  direction: Direction,
+}
+
+export type ListType = 'LIST' | 'GRID'
+
+export interface DynamicListInterface extends BeagleGridViewInterface, BeagleGridViewInterface{
+  type: ListType,
+}
```

Comment:
On TypeScript, we are able to create [String Enums](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums), so would be a cleaner approach to use on the application, avoiding some casts using `as` and is easier to do maintenance and for new types of dynamic list. Other benefits are the ability to quickly find all references to the list type on the whole project structure and it is also really quicker to write, because using the auto-complete of your editor you do not have to worry about misspelling the string or validating it.

I suggest implement this (also valid for Direction) like this:

```suggestion
export enum ListType {
  Grid='GRID',
  List='LIST'
}

export enum ListDirection {
  Horizontal='HORIZONTAL',
  Vertical='VERTICAL'
}

export interface BeagleGridViewInterface extends DynamicListBaseInterface {  
  numColumns?: number,
}

export interface BeagleListViewInterface extends DynamicListBaseInterface{
  direction: ListDirection,
}

export interface DynamicListInterface extends BeagleGridViewInterface, BeagleGridViewInterface{
  type: ListType,
}
```